openvdb/points/AttributeArray.h

```diff
@@ -836,6 +1019,92 @@ UnitVecCodec::encode(const math::Vec3<T>& val, StorageType& data)
 }
 
 
+////////////////////////////////////////
+
+// AttributeArray implementation
+
+#if OPENVDB_ABI_VERSION_NUMBER >= 6
+template <typename IterT>
+void AttributeArray::copyValuesUnsafe(const AttributeArray& sourceArray, const IterT& iter,
+    bool rangeChecking/* = false*/)
+{
+    // ensure both arrays have float-float or integer-integer value types
+    assert(sourceArray.valueTypeIsFloatingPoint() != this->valueTypeIsFloatingPoint());
+    // ensure both arrays have been loaded from disk (if delay-loaded)
+    assert(sourceArray.isDataLoaded() && this->isDataLoaded());
+    // ensure storage size * stride matches on both arrays
+    assert(this->storageTypeSize()*this->stride() == sourceArray.storageTypeSize()*sourceArray.stride());
+
+    const size_t bytes(sourceArray.storageTypeSize()*sourceArray.stride());
+    const char* const sourceBuffer = sourceArray.dataAsByteArray();
+    char* const targetBuffer = this->dataAsByteArray();
+    assert(sourceBuffer && targetBuffer);
+
+    if (rangeChecking && this->isUniform()) {
+        OPENVDB_THROW(IndexError, "Cannot copy array data as target array is uniform.");
+    }
```

Comment:
can we move ```this->expand();``` from copyValues into here so this is never a concern?

Reply:
On reflection, I'm not going to make this change as without it, it becomes possible to call this method with a uniform target array and an iterator that only generates zero indices to avoid having to go through type instantiation.

I have however documented this use case and added an additional unit test to ensure this behavior is verified.